src/Forms/GridField/GridFieldGroupDeleteAction.php

```diff
@@ -86,21 +87,21 @@ public function handleAction(GridField $gridField, $actionName, $arguments, $dat
      */
     protected function canUnlink($record)
     {
+        // If the record isn't the current member or isn't an admin, don't allow unlinking.
         $currentUser = Security::getCurrentUser();
-        if ($currentUser
-            && $record instanceof Member
-            && (int)$record->ID === (int)$currentUser->ID
-            && Permission::checkMember($record, 'ADMIN')
+        if (!$currentUser
+            || !($record instanceof Member)
+            || (int)$record->ID !== (int)$currentUser->ID
+            || !Permission::checkMember($record, 'ADMIN')
         ) {
-            $adminGroups = array_intersect(
-                $record->Groups()->column() ?? [],
-                Permission::get_groups_by_permission('ADMIN')->column()
-            );
-
-            if (count($adminGroups ?? []) === 1 && array_search($this->groupID, $adminGroups ?? []) !== false) {
-                return false;
-            }
+            return false;
+        }
+        // If there is exactly one admin group and it's THIS group, don't allow unlinking.
+        $adminGroups = $record->Groups()->filterByList(Permission::get_groups_by_permission('ADMIN'));
+        if ($adminGroups->count() === 1 && $adminGroups->filter(['ID' => $this->groupID])->exists()) {
```

Comment:
```suggestion
        if ($adminGroups->count() === 1 && $adminGroups->first()->ID === $this->groupID) {
```
Slightly easier to read, I'm assuming the ->exists() check isn't necessary

Reply:
`exists()` is more efficient than `first()->ID` because `first()`:
1. Selects ALL the data for this record in the database
2. Hydrates a `DataObject` in PHP

whereas `exists()`:
1. Checks if there is a row in the database and returns a boolean.... and that's it.